### PR TITLE
loyalsoldier-geoip 202407250105 (new formula)

### PR DESCRIPTION
**Pull request by Loyalsoldier**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/l/loyalsoldier-geoip.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        merge CIDRs from local file and only print IPv4 CIDRs to stdout:
+        $ cat redundant-cidr.txt | geoip merge -t ipv4
+      #{"  "}
+        merge CIDRs from remote URL and print to stdout:
+        $ curl -s https://core.telegram.org/resources/cidr.txt | geoip merge
+      #{"  "}
+        merge CIDRs from remote URL and only print IPv6 CIDRs to stdout:
+        $ curl -s https://core.telegram.org/resources/cidr.txt | geoip merge -t ipv6
+    EOS
+  end
```
Comment (SMillerDev) [inline Formula/l/loyalsoldier-geoip.rb:32]: This is not Homebrew specific and should be in upstream documentation

**Comment by SMillerDev** [comment]
> We frown on authors submitting their own work unless it is very popular.

https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff

File: Formula/l/loyalsoldier-geoip.rb (hunk 1 above)
Comment (Loyalsoldier) [reply Formula/l/loyalsoldier-geoip.rb]: Removed.

**Comment by Loyalsoldier** [comment]
> > We frown on authors submitting their own work unless it is very popular.
> 
> https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff

I think 3.4K stars and 600 forks should be popular enough 🙊

**Comment by Loyalsoldier** [comment]
Any progress?

**Comment by p-linnane** [comment]
Hi @Loyalsoldier. Thanks for your patience. This looks good. Can you please remove `conflicts_with` from both formulae? Including `geoip` here would cause it to be rebuilt, which we don't want. We can handle the conflict in a follow up PR. Once that change is made I will merge this, since I agree that the stars and forks are enough for inclusion.

**Comment by Loyalsoldier** [comment]
@p-linnane  Done.

BTW, is there anyway to update the binary just base on the code changes in master branch, not the release tag? This project releases a tag weekly, but the binary would be the same at most time.

**Comment by p-linnane** [comment]
Thanks. Will merge when CI is done.

>BTW, is there anyway to update the binary just base on the code changes in master branch, not the release tag?

That is what [`head`](https://docs.brew.sh/Formula-Cookbook#unstable-versions-head) is for. Users can install either the stable tag, or the latest commit since it's specified in the formula.

**Comment by Loyalsoldier** [comment]
@p-linnane What I mean is that tag will be release weekly (to publish the geoip database) on the same commit even if code is unchanged, it's unnecessary to auto update it based on tag in homebrew because the binary(CLI) is the same at most time.

**Comment by p-linnane** [comment]
No, that kind of release model does not work well in Homebrew. You may want to host this [in your own tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap).

**Pull request closed without merging**